crypto/pkcs7/bio/md.c

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <errno.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#include <openssl/pkcs7.h>
+#include "../../internal.h"
+#include "../crypto/bio/internal.h"
+#include "../internal.h"
+
+// BIO_put and BIO_get both add to the digest, BIO_gets returns the digest
```

Comment:
Was this a leaked note? It doesn't seem like `BIO_f_md` has a puts method or am I overlooking something

Reply:
hm, should be "md_write and md_read both contribute to the digest, md_gets returns digest contents"